### PR TITLE
T41 add Unit test for get category list and get detail category

### PR DESCRIPTION
**Pull request by PhaciusPhat**
unit test when getting valid categories successfully
image
unit test when getting a valid category 
image
unit test when getting a category with not found Id 
image

### CODE REVIEW

**Comment by PhaciusPhat** [comment]
create category with parentId and without parentId
image
create category with invalid parentId
image

File: product/src/test/java/com/yas/product/controller/CategoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        );
+        var categoryCaptor = ArgumentCaptor.forClass(Category.class);
+        Category savedCategory = mock(Category.class);
+        when(categoryRepository.findById(categoryPostVm.parentId())).thenReturn(Optional.of(category));
+        when(categoryRepository.saveAndFlush(categoryCaptor.capture())).thenReturn(savedCategory);
+        UriComponentsBuilder newUriComponentsBuilder = mock(UriComponentsBuilder.class);
+        UriComponents uriComponents = mock(UriComponents.class);
+        when(uriComponentsBuilder.replacePath("/categories/{id}")).thenReturn(newUriComponentsBuilder);
+        when(newUriComponentsBuilder.buildAndExpand(savedCategory.getId())).thenReturn(uriComponents);
+
```
Comment (thiennn) [inline product/src/test/java/com/yas/product/controller/CategoryControllerTest.java:144]: please help remove 1 blank line

File: product/src/test/java/com/yas/product/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+        ResponseEntity<CategoryGetDetailVm> actual = categoryController.createCategory(categoryPostVm
+                , uriComponentsBuilder, principal);
+
```
Comment (thiennn) [inline product/src/test/java/com/yas/product/controller/CategoryControllerTest.java:148]: please help remove 1 blank line

File: product/src/test/java/com/yas/product/controller/CategoryControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        categoryPostVm = new CategoryPostVm(
+                "name",
+                "slug",
+                "description",
+                null,
+                "",
+                "",
+                (short) 0
+        );
+        // create captor data when we send it to save method
```
Comment (thiennn) [inline product/src/test/java/com/yas/product/controller/CategoryControllerTest.java:98]: we shouldn't add comment for obvious code

**Review by thiennn — CHANGES_REQUESTED** [review]

**Review by thiennn — APPROVED** [review]

**Pull request merged**